File: uri_exercicios_em_python/begginner/animal.py
def find_animal(a, b, c):
    animals = {
        "vertebrado": {
            "ave": {
                "carnivoro": "aguia",
                "onivoro": "pomba"
            },
            "mamifero": {
                "onivoro": "homem",
                "herbivoro": "vaca"
            }
        },
        "invertebrado": {
            "inseto": {
                "hematofago": "pulga",
                "herbrivoro": "lagarta"
            },
            "anelideo": {
                "hematofago": "sanguessuga",
                "onivoro": "minhoca"
            }
        }
    }

    return animals[a][b][c]


x = input()
y = input()
z = input()

print(find_animal(x, y, z))

File: beecrowd_exercicios_em_Cpp/beginner/animal.cpp
#include <iostream>
#include <map>
#include <string>
using namespace std;

map <string,map<string,map<string,string>>> criar_map(map <string,map<string,map<string,string>>> animals){
        animals["vertebrado"]["ave"]["carnivoro"] = "aguia";
        animals["vertebrado"]["ave"]["onivoro"] = "pomba";
        animals["vertebrado"]["mamifero"]["onivoro"] = "homem";
        animals["vertebrado"]["mamifero"]["herbivoro"] = "vaca";
        animals["invertebrado"]["inseto"]["herbivoro"] = "lagarta";
        animals["invertebrado"]["inseto"]["hematofago"] = "pulga";
        animals["invertebrado"]["anelideo"]["hematofago"] = "sanguessuga";
        animals["invertebrado"]["anelideo"]["onivoro"] = "minhoca";

    return animals;
}


int main(){

    map <string,map<string,map<string,string>>> animals {};
    animals = criar_map(animals);

    string a,b,c;

    cin >> a >> b >> c;

    cout << animals[a][b][c] << endl;

    return 0;
}